src/plugins/csp_configuration_provider/server/provider.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchClient } from '../../../../src/core/server';
+
+import { CspClient } from './types';
+
+export class OpenSearchCspClient implements CspClient {
+  private client: OpenSearchClient;
+
+  constructor(inputOpenSearchClient: OpenSearchClient) {
+    this.client = inputOpenSearchClient;
+  }
+
+  async exists(configurationName: string): Promise<boolean> {
```

Comment:
Do we need this method? It seems that `get` can cover `exists`.

Reply:
If we only use `get`, it will throw exception when there is no this index.

Also it will generate an error log in OSD log.

```
server    log   [22:47:21.193] [error][data][opensearch] [index_not_found_exception]: no such index [.opensearch_dashboards_config]
```

This will cause confusion and waste system resource to handle exception.

As a contrast, `exists` will handle the same case smoothly.